pkg/controllers/state/statenode.go

```diff
@@ -220,7 +219,7 @@ func (in *StateNode) Taints() []v1.Taint {
 	// We can assume that all nodes will have this label and no back-propagation will be required once we hit v1
 	if !in.Registered() && !in.Initialized() && in.NodeClaim != nil {
```

Comment:
```suggestion
	if (!in.Registered() && !in.Initialized() && in.NodeClaim != nil) || in.Node == nil {
```

Should this be the handling? If it's not initialized or registered and the nodeclaim is nil or if the node doesn't exist yet.